cassiopeia/solver/solver_utilities.py

```diff
@@ -165,15 +166,42 @@ def collapse_tree(
     # Calls helper function on root, passing in the mapping dictionary
     collapse_edges(T, root, node_to_characters)
 
+def collapse_unifurcations(tree: ete3.Tree):
+    """Collapse unifurcations.
+
+    Collapse all unifurcations in the tree, namely any node with only one child
+    should be removed and all children should be connected to the parent node.
+
+    Args:
+        tree: tree to be collapsed
+
+    Returns:
+        A collapsed tree.
+    """
+
+    collapse_fn = lambda x: (len(x.children) == 1)
+
+    collapsed_tree = tree.copy()
+    to_collapse = [n for n in collapsed_tree.traverse() if collapse_fn(n)]
+
+    for n in to_collapse:
+        n.delete()
+
+    return collapsed_tree
```

Comment:
Are the children of the nodes attached to the parent of the deleted node?

Reply:
Yes, this is done whenever you delete a given node: http://etetoolkit.org/docs/latest/reference/reference_tree.html#ete3.TreeNode.delete